README.md

```diff
@@ -1,2 +1,307 @@
+
 # pubsub
-Publish and subscribe functionality within a single process in Go.
+    import "github.com/juju/pubsub"
+
+Package pubsub provides publish and subscribe functionality within a single process.
+
+The core aspect of the pubsub package is the `Hub`. The hub just provides
+two methods:
+* Publish
+* Subscribe
+
+A message as far as a hub is concerned is defined by a topic, and a data
+blob. All subscribers that match the published topic are notified, and have
```

Comment:
"All  subscribers that have a subscription which matches the topic for a message are notified,..."